transport.go

```diff
@@ -2,20 +2,56 @@ package transport
 
 import (
 	"context"
+	"io"
 	"net"
 	"time"
 
 	logging "github.com/ipfs/go-log"
+	smux "github.com/libp2p/go-stream-muxer"
 	ma "github.com/multiformats/go-multiaddr"
-	manet "github.com/multiformats/go-multiaddr-net"
 )
 
 var log = logging.Logger("transport")
 
 // Conn is an extension of the net.Conn interface that provides multiaddr
 // information, and an accessor for the transport used to create the conn
 type Conn interface {
-	manet.Conn
+	io.Closer
+
+	RemoteAddr() net.Addr
+	RemoteMultiaddr() ma.Multiaddr
+
+	LocalAddr() net.Addr
+	LocalMultiaddr() ma.Multiaddr
+
+	Transport() Transport
+}
+
+// A SingleStreamConn is a connection that provides a single channel between two endpoints
+// e.g. a TCP connection
+type SingleStreamConn interface {
+	Conn
+
+	io.Reader
+	io.Writer
+
+	SetDeadline(time.Time) error
+	SetReadDeadline(time.Time) error
+	SetWriteDeadline(time.Time) error
+}
+
+// A MultiStreamConn is a connection that supports transport-level stream multiplexing.
+// e.g. a QUIC connection
+// The MultiStreamConn combines the smux.Conn and the Conn interface
+// (unfortunately Go still doesn't allow duplicate interface methods...)
+type MultiStreamConn interface {
```

Comment:
maybe `MuxingConn` ?

Reply:
I'd be fine with that, but then we probably should also rename the `SingleStreamConn` as well. Any suggestions for that?

Reply:
hrm... maybe `DirectConn` or `BaseConn`? @Stebalien ?

Reply:
To me, `BaseConn` implies that all other conns inherit from it and `DirectConn` implies that the connection is direct (which is true for all connections).

How about:

* `StreamingConn` (like `Muxing`)
* `BasicConn` (doesn't imply that it's the "base", just a minimal feature set).

(I still kind of like `DuplexConn`/`MultiplexConn` or `DuplexedConn`/`MultiplexedConn` but those are kind of long).

Reply:
@marten-seemann

So, I'd like to make progress on this set of PRs (both for QUIC and because they're holding up some other updates) but neither @whyrusleeping nor I really care about the names of these connections... we just don't like MultiStreamConn because we already have a concept of multistreams (and they're completely unrelated).

So... you decide (roll dice if you want). Just not MultiStreamConn.

Reply:
I picked `DuplexConn` and `MultiplexConn`, and updated all PRs where those were used.